src/main/scala/io/iohk/ethereum/network/p2p/Message.scala

```diff
@@ -17,6 +17,8 @@ trait MessageSerializable extends Message {
 
   def toBytes: Array[Byte]
 
+  def underlyingMsg: Message
```

Comment:
That's right, the messages transmitted to be sent to the Network layer should always be `MessageSerializable`, and the use of the abstract classes for obtaining them always happens outside the network.